### PR TITLE
[Storage] BREAKING CHANGE: `az storage share close-handle`: Remove `--marker` which is not supported by sdk

### PR DESCRIPTION
**Pull request by calvinhzy**
**Related command**

**Description**

`az storage share list-handle\close-handle`: migrate to track2 sdk 

**Testing Guide**

    # To reproduce the prerequisite steps:
    # 1. Create a file share under a storage account/ resource group as specified below
    # 2. Upload Book1.csv to the root dir of the file share, add a dir called dir1 and upload another file testjson.json
    # 3. Create a Python file to open both files and sleep so the file handles are held
    # 4. Create two VMs and ssh to them
    # 5. Mount the file share using the script in the connect tab and run the Python file
`az storage share list-handle -n {share_name} --recursive --path {file_path}`
`az storage share close-handle --name {share_name} --path  {file_path} --recursive --close-all`

**History Notes**

[Storage] BREAKING CHANGE: `az storage share close-handle`: Remove `--marker` which is not supported by sdk

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Storage

File: src/azure-cli/azure/cli/command_modules/storage/operations/fileshare.py [hunk 1]
```diff
@@ -93,3 +93,57 @@ def get_share_stats(client, timeout=None, **kwargs):
...
+
+    generator = client.list_handles(results_per_page=num_results, **kwargs)
+    pages = generator.by_page(continuation_token=marker)  # SharePropertiesPaged
+    result = list_generator(pages=pages, num_results=num_results)
+
+    if pages.continuation_token:
+        next_marker = {"nextMarker": pages.continuation_token}
+        result.append(next_marker)
+
+    return result
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/operations/fileshare.py:125]:
```suggestion
    return {"item": result, "nextMarker": pages.continuation_token}
```

File: src/azure-cli/azure/cli/command_modules/storage/_transformers.py [hunk 2]
```diff
@@ -428,3 +428,17 @@ def transform_share_file_json_output(result):
...
+    if len(result) > 1 and isinstance(result[-1], dict):
+        new_result["items"] = result[:-1]
+        new_result["nextMarker"] = result[-1]["nextMarker"]
+    else:
+        new_result["items"] = result
+        new_result["nextMarker"] = None
+    for item in new_result["items"]:
+        item["handleId"] = item.id
+        delattr(item, "id")
+    return new_result
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_transformers.py:444]:
```suggestion
def transform_share_list_handle(result):
    for item in result["items"]:
        item["handleId"] = item.id
        delattr(item, "id")
    return result
```

Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/storage/_transformers.py:444]: To make code concise

File: src/azure-cli/azure/cli/command_modules/storage/operations/fileshare.py [hunk 3]
```diff
@@ -93,3 +93,57 @@ def get_share_stats(client, timeout=None, **kwargs):
...
+        total_path = directory_path + '/' + file_name if directory_path else file_name
+        dir_client = client.get_directory_client(directory_path=total_path)
+        if not dir_client.exists():
+            dir_client = client.get_directory_client(directory_path=directory_path)
+            client = dir_client.get_file_client(file_name=file_name)
+            kwargs.pop("recursive")
+        else:
+            client = dir_client
+    else:
+        client = dir_client
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/operations/fileshare.py:144]: Seems to be the same with `list_handle()`. Can we extract a common function for this?

File: src/azure-cli/azure/cli/command_modules/storage/operations/fileshare.py [hunk 4]
```diff
@@ -93,3 +93,57 @@ def get_share_stats(client, timeout=None, **kwargs):
 def set_share_metadata(client, metadata=None, timeout=None, **kwargs):
     client.set_share_metadata(metadata=metadata, timeout=timeout, **kwargs)
     return True
+
+
+def list_handle(client, marker, num_results, **kwargs):
+    from ..track2_util import list_generator
+    directory_path = kwargs.pop("directory_name", None)
+    if directory_path and directory_path.startswith('./'):
+        directory_path = directory_path.replace('./', '', 1)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/operations/fileshare.py:102]: Do we really have such cases to pass in `./dir/file` as `--path`? What if customers provide `/dir/file`, should we escape the first `/`?

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**